sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -944,25 +944,28 @@ def test_neighbors_metrics(n_samples=20, n_features=3,
     for metric, metric_params in metrics:
         results = []
```

Comment:
If we made results a dict keyed by algorithm, I think this test and its failings would be much clearer.